site/content/entry/citra-progress-report-2017-august.md

```diff
@@ -0,0 +1,155 @@
+recently, no one gave any attention to the software backend, as OpenGL was faster,
+made better use of the GPU, and allowed things such as texture forwarding for
+higher resolution rendering. But there is one thing on which the OpenGL backend
+falls flat on its face; accuracy.
+
+But, [wwylele](https://github.com/wwylele) has just revived the software renderer
+(inspired by an attempt by [Subv](https://github.com/Subv) made prior), adding
+almost all of the features the hardware renderer had recieved over the years.
+In fact, every addition to the hardware renderer has been given an equivalent
+in software:
+
+{{% table %}}
+|      Feature      |                        Hardware                       |                                                   Software                                                   |
+|:-----------------:|:-----------------------------------------------------:|:------------------------------------------------------------------------------------------------------------:|
+| Fragment Lighting | [#1264](https://github.com/citra-emu/citra/pull/1264) |  [#2766](https://github.com/citra-emu/citra/pull/2766) [#2822](https://github.com/citra-emu/citra/pull/2822) |
+|   Spot Lighting   | [#2727](https://github.com/citra-emu/citra/pull/2727) |                             [#2871](https://github.com/citra-emu/citra/pull/2871)                            |
+|  Geometric Factor | [#2776](https://github.com/citra-emu/citra/pull/2776) |                             [#2872](https://github.com/citra-emu/citra/pull/2872)                            |
+|    Bump Mapping   | [#2762](https://github.com/citra-emu/citra/pull/2762) |                             [#2891](https://github.com/citra-emu/citra/pull/2891)                            |
+{{% /table %}}
+
+Although most users won't benefit much from this in the present, it is important
+to have a complete software implementation of the 3DS' GPU so that Citra can be
+used as a stable, working, and complete reference implementation in the future,
+when obtaining a working real console may be much more difficult.
+
+## [Handle Invalid Filenames When Renaming](https://github.com/citra-emu/citra/pull/2850) by [j-selby](https://github.com/j-selby)
+
+Citra emulates the 3DS system services at a high level of emulation, or HLE for
+short. What this means is that every time a 3DS application or game running in
+Citra makes a request to the 3DS System Software, Citra captures the request and
+tries to translate it into its PC equivalent, rather then running the 3DS System
+Software directly.
+
+For example, if a game makes a call to [`FS:OpenFile`](https://www.3dbrew.org/wiki/FS:OpenFile),
+Citra in turn calls the operating system's file opening function
+([`CreateFile()`](https://msdn.microsoft.com/en-us/library/windows/desktop/aa363858(v=vs.85).aspx) on Windows,
+[`FileHandle()`](https://developer.apple.com/documentation/foundation/filehandle) on macOS,
```

Comment:
Citra uses`_wfopen_s` on Windows/ `open()` on macOS - <https://github.com/citra-emu/citra/blob/master/src/common/file_util.cpp#L832>

Reply:
Not a big deal. `_wfopen_s` itself essentially calls system API like `CreateFile`